src/components/Envoy/EnvoyModal.tsx

```diff
@@ -0,0 +1,234 @@
+        Cancel
+      </Button>
+    ]}
+  >
+    <EnvoyDetail namespace={namespace} workload={workload} onClose={onClose} />
+  </Modal>
+);
+
+export const Loading = () => (
+  <EmptyState variant={EmptyStateVariant.full}>
+    <EmptyStateIcon variant="container" component={Spinner} />
+    <Title size="lg" headingLevel="h4">
+      Loading...
+    </Title>
+  </EmptyState>
+);
+
+class EnvoyDetail extends React.Component<EnvoyDetailProps, EnvoyDetailState> {
+  aceEditorRef: React.RefObject<AceEditor>;
+
+  constructor(props: EnvoyDetailProps) {
+    super(props);
+    this.aceEditorRef = React.createRef();
+    this.state = {
+      config: 'loading...',
+      resource: 'all',
+      fetch: true,
+      pod: this.sortedPods()[0]
+    };
+  }
+
+  componentDidMount() {
+    this.fetchContent();
+  }
+
+  componentDidUpdate() {
+    if (this.state.fetch) {
+      this.fetchContent();
+    }
+  }
+
+  sortedPods = (): Pod[] => {
+    return this.props.workload.pods.sort((p1: Pod, p2: Pod) => (p1.name >= p2.name ? 1 : -1));
+  };
+
+  setPod = (podName: string) => {
+    const podIdx: number = +podName;
+    const targetPod: Pod = this.sortedPods()[podIdx];
+    if (targetPod.name !== this.state.pod.name) {
+      this.setState({
+        fetch: true,
+        pod: targetPod
+      });
+    }
+  };
+
+  setResource = (resource: string) => {
+    const resourceIdx: number = +resource;
+    const targetResource: string = resources[resourceIdx];
+    if (targetResource !== this.state.resource) {
+      this.setState({
+        fetch: true,
+        resource: targetResource
+      });
+    }
+  };
+
+  fetchContent = () => {
+    if (this.state.resource === 'all') {
+      this.fetchEnvoyProxy();
+    } else {
+      this.fetchEnvoyProxyResourceEntries();
+    }
+  };
+
+  fetchEnvoyProxy = () => {
+    API.getPodEnvoyProxy(this.props.namespace, this.state.pod.name)
+      .then(resultEnvoyProxy => {
+        this.setState({
+          config: resultEnvoyProxy.data,
+          fetch: false
+        });
+      })
+      .catch(error => {
+        this.props.onClose();
+        AlertUtils.addError(`Could not fetch envoy config for ${this.state.pod.name}.`, error);
+      });
+  };
+
+  fetchEnvoyProxyResourceEntries = () => {
+    API.getPodEnvoyProxyResourceEntries(this.props.namespace, this.state.pod.name, this.state.resource)
+      .then(resultEnvoyProxy => {
+        this.setState({
+          config: resultEnvoyProxy.data,
+          fetch: false
+        });
+      })
+      .catch(error => {
+        this.props.onClose();
+        AlertUtils.addError(
+          `Could not fetch envoy config ${this.state.resource} entries for ${this.state.pod.name}.`,
+          error
+        );
+      });
+  };
+
+  render() {
+    const builder = SummaryWriterBuilder(this.state.resource, this.state.config);
+    const SummaryWriterComp = builder[0];
+    const summaryWriter = builder[1];
+    return (
+      <Stack gutter={GutterSize.sm}>
```

Comment:
Also, using the WorkloadWizard would help to not need the Stack component.
I think I remove it (or delegate it) to any of the subwizards to render it.

Perhaps you can take a look to the Service Wizards as well.

Reply:
I think we need the stack anyway since every modal content needs a grid to distribute the content.
I see that in the WorkloadWizard there is nothing in the content just a plain text.
Perhaps I missed something. I could use a clarification here.